### PR TITLE
Cleanup artifacts to build and setup apiextensions-apiserver

### PR DESCRIPTION
**Pull request by nikhita**
These files were added in https://github.com/kubernetes/kubernetes/commit/6a74988f96ea3df0fe9d28e27a29d398083110de when apiextensions-apiserver was required to be setup using this script because it wasn't embedded in: https://github.com/nikhita/gsoc-meta-k8s/blob/master/notes/set-kube-apiextensions-server.sh.

We don't need it today so this PR removes it.

/kind cleanup
/assign @sttts 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/artifacts/customresource-01/noxu-resource-definition.yaml [hunk 1]
```diff
@@ -1,13 +0,0 @@
...
-metadata:
-  name: noxus.mygroup.example.com
-spec:
-  group: mygroup.example.com
-  version: v1alpha1
-  scope: Namespaced
-  names:
-    plural: noxus
-    singular: noxu
-    kind: Noxu
```
Comment (yue9944882) [inline staging/src/k8s.io/apiextensions-apiserver/artifacts/customresource-01/noxu-resource-definition.yaml]: these noxus finished their k/k journey 🙂

Comment (nikhita) [reply staging/src/k8s.io/apiextensions-apiserver/artifacts/customresource-01/noxu-resource-definition.yaml]: Haha, no they are still being used in integration tests :smile:

**Comment by nikhita** [comment]
Rebased. @sttts Ptal.

**Pull request merged**